File: dependency-injection-with-dagger2-qualifier/app/src/main/java/com/example/dependencyinjectionwithdagger2/AppComponent.java
package com.example.dependencyinjectionwithdagger2;

import android.content.SharedPreferences;

import javax.inject.Singleton;

import dagger.Component;
import okhttp3.OkHttpClient;
import retrofit2.Retrofit;

@Singleton
@Component(modules={AppModule.class, NetModule.class})
public interface AppComponent {

    void inject(MyActivity myActivity);

}

File: dagger2-scopes-and-subcomponents2/app/src/main/java/com/example/dagger2scopesandsubcomponentspart2/AppComponent.kt
package com.example.dagger2scopesandsubcomponentspart2

import dagger.Component

//@ApplicationScope
//@Component(modules = [AppModule::class])
//interface AppComponent {
//    fun getWarrior(): Warrior
//}


@ApplicationScope
@Component(modules = [AppModule::class])
interface AppComponent {
    fun warriorScreenComponent(warriorScreenModule: WarriorScreenModule) : WarriorScreenComponent
}